ReactiveCocoa/Swift/Signal.swift

```diff
@@ -1031,6 +1031,39 @@ extension SignalType {
 	}
 }
 
+extension SignalType where Value: Hashable {
+	/// Forwards only those values from `self` that are unique across the set of
+	/// all values that have been seen.
+	/// Note: This causes values that are forwarded to be retained to check for
+	/// uniquness.
+	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
+	public func uniqueValues() -> Signal<Value, Error> {
+		return Signal { observer in
+			let seenValues = Atomic<Set<Value>>(Set<Value>())
+			
+			return self
+				.filter { value in
+					return seenValues.withValue { set in
+						!set.contains(value)
+					}
+				}
+				.observe { event in
+					switch event {
+					case let .Next(value):
+						seenValues.modify { set in
```

Comment:
I thought that `var [name] in` was correct syntax as well, but it looks like its not valid.

<img width="998" alt="screen shot 2016-02-18 at 8 57 57 pm" src="https://cloud.githubusercontent.com/assets/18454/13166923/68f36242-d682-11e5-8000-1a1cde4cc55c.png">